### ISSUE TITLE
NiftiMapsMasker function should have an extraction strategy parameter

### ISSUE THREAD

**Issue by michaelbernier** [nilearn/nilearn#1578]
Hello, 

It would be useful to incorporate extraction strategies when using NiftiMaskers function (like NiftiMapsMasker). Right now it seems to compute the mean signal of every regions, but for some applications, I would need the max, min, non-zero mean, std, sum, etc. Or is there already a way to accomplish this?

Good job to the developers BTW!

**Comment by bthirion** [comment]
My advice for that would be extract the times series using a mask per roi and compute the quantity you wish with the adequate numpy function. Allowing all possible operations in the API leads us into feature creep.

**Comment by michaelbernier** [comment]
Good point, It is indeed feasible in separate steps.. Thanks for the quick reply!

**Comment by dPys** [comment]
Second this suggestion @michaelbernier. Maybe like a mutual information (using http://scikit-learn.org/stable/modules/generated/sklearn.metrics.normalized_mutual_info_score.html ?) across voxels in the mask as opposed to just grabbing the mean?

[https://ieeexplore.ieee.org/document/6905833/](https://ieeexplore.ieee.org/document/6905833/)

**Comment by bthirion** [comment]
Stalled. Closing.
